apis/clusters/v1beta1/opensearch_webhook_test.go

```diff
@@ -0,0 +1,315 @@
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/instaclustr/operator/pkg/models"
+)
+
+var _ = Describe("Kafka Controller", Ordered, func() {
+	openSearchManifest := OpenSearch{}
+	testOpenSearchManifest := OpenSearch{}
+
+	It("Reading kafka manifest", func() {
+		yfile, err := os.ReadFile("../../../controllers/clusters/datatest/opensearch_v1beta1.yaml")
+		Expect(err).Should(Succeed())
+
+		err = yaml.Unmarshal(yfile, &openSearchManifest)
+		Expect(err).Should(Succeed())
+		openSearchManifest.Spec.TwoFactorDelete = []*TwoFactorDelete{{Email: "emailTEST", Phone: "phoneTEST"}}
+		testOpenSearchManifest = openSearchManifest
+	})
+
+	ctx := context.Background()
+
+	When("apply a OpenSearch manifest", func() {
+		It("should test OpenSearch creation flow", func() {
+			testOpenSearchManifest.Spec.TwoFactorDelete = []*TwoFactorDelete{openSearchManifest.Spec.TwoFactorDelete[0], openSearchManifest.Spec.TwoFactorDelete[0]}
+			Expect(k8sClient.Create(ctx, &testOpenSearchManifest)).ShouldNot(Succeed())
+			testOpenSearchManifest.Spec.TwoFactorDelete = openSearchManifest.Spec.TwoFactorDelete
+
+			testOpenSearchManifest.Spec.SLATier = "some SLATier that is not supported"
+			Expect(k8sClient.Create(ctx, &testOpenSearchManifest)).ShouldNot(Succeed())
+			testOpenSearchManifest.Spec.SLATier = openSearchManifest.Spec.SLATier
+
+			testOpenSearchManifest.Spec.DataCentres = []*OpenSearchDataCentre{}
+			Expect(k8sClient.Create(ctx, &testOpenSearchManifest)).ShouldNot(Succeed())
+			testOpenSearchManifest.Spec.DataCentres = openSearchManifest.Spec.DataCentres
+
+			prevDataNode := openSearchManifest.Spec.DataNodes[0]
+			testOpenSearchManifest.Spec.DataNodes = []*OpenSearchDataNodes{}
+			Expect(k8sClient.Create(ctx, &testOpenSearchManifest)).ShouldNot(Succeed())
+			testOpenSearchManifest.Spec.DataNodes = []*OpenSearchDataNodes{prevDataNode}
+
+			prevDC := *openSearchManifest.Spec.DataCentres[0]
+			testOpenSearchManifest.Spec.DataCentres[0].CloudProvider = "test"
+			Expect(k8sClient.Create(ctx, &testOpenSearchManifest)).ShouldNot(Succeed())
+
+			testOpenSearchManifest.Spec.DataCentres[0].CloudProvider = "AWS_VPC"
+			testOpenSearchManifest.Spec.DataCentres[0].Region = "test"
+			Expect(k8sClient.Create(ctx, &testOpenSearchManifest)).ShouldNot(Succeed())
+
+			testOpenSearchManifest.Spec.DataCentres[0].CloudProvider = "AZURE_AZ"
+			Expect(k8sClient.Create(ctx, &testOpenSearchManifest)).ShouldNot(Succeed())
+
+			testOpenSearchManifest.Spec.DataCentres[0].CloudProvider = "GCP"
+			Expect(k8sClient.Create(ctx, &testOpenSearchManifest)).ShouldNot(Succeed())
+			testOpenSearchManifest.Spec.DataCentres[0] = &prevDC
+
+			prevStringValue := openSearchManifest.Spec.DataCentres[0].ProviderAccountName
+			testOpenSearchManifest.Spec.DataCentres[0].ProviderAccountName = models.DefaultAccountName
+			Expect(k8sClient.Create(ctx, &testOpenSearchManifest)).ShouldNot(Succeed())
+			testOpenSearchManifest.Spec.DataCentres[0].ProviderAccountName = prevStringValue
```

Comment:
Do we plan to add tests for RIYOA accounts?

RIYOA - run in your own account. It means, that customer will run Instaclustr Products using their own AWS/AZURE/GCP accounts.

Reply:
this test case is made for ensure that  cloudProviderSettings is available only for RIYOA accounts